is-valid-turn.js

```diff
@@ -0,0 +1,89 @@
+"use strict";
+const isValidMove = require("./is-valid-move");
+const makeMove = require("./make-move");
+const _ = require("lodash");
+
+function isValidTurn(gameState, diceRoll, proposedMoves) {
+    function getAceyDeucey() {
+        const aceyDeucey = {};
+        diceRoll.forEach(function(roll, index) {
+            if (roll === 1 && !aceyDeucey.hasOwnProperty("oneLocation")) {
+                aceyDeucey.oneLocation = index; 
+            } else if (roll === 2 && !aceyDeucey.hasOwnProperty("twoLocation")) {
+                aceyDeucey.twoLocation = index;
+            } else if (!aceyDeucey.hasOwnProperty("doublesLocation")) {
+                aceyDeucey.doublesLocation = index;
+            } else {
+                aceyDeucey.isAceyDeucey = false;
+            }
+        });
+        return aceyDeucey;
+    }
+    
+    function proposedMovesMatchDice() {
+        if (diceRoll.length > 2) {
+                      
+            const aceyDeucey = getAceyDeucey();
+            if (aceyDeucey.isAceyDeucey === false) {
+                return false;
+            }
+
+            if (
+                !(proposedMoves[0].numberOfSpaces === 1 && proposedMoves[1].numberOfSpaces === 2) &&
+                !(proposedMoves[0].numberOfSpaces === 2 && proposedMoves[1].numberOfSpaces === 1)
+            ) {
+                return false;
+            }
+            
+            const sortedAceyDeuceyRoll = _.sortBy(diceRoll);
+            for (let i = 0; i < 3; i++) {
```

Comment:
This confused me a bit, because I kept looking to see where you were using `i`. Then I realized you just wanted to take the same action three times. What if you did the following instead?

```js
_(3)
    .range()
    .forEach(() => sortedAceyDeuceyRoll.push(diceRoll[aceyDeucey.doublesLocation]));
```